src/promptflow-devkit/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -415,17 +444,25 @@ def visualize(self, runs: Union[str, Run, List[str], List[Run]], **kwargs) -> No
         if not isinstance(runs, list):
             runs = [runs]
 
-        validated_runs = []
+        validated_runs: List[Run] = []
         for run in runs:
             run_name = Run._validate_and_return_run_name(run)
             validated_runs.append(self.get(name=run_name))
 
         html_path = kwargs.pop("html_path", None)
-        try:
-            self._visualize(validated_runs, html_path=html_path)
-        except InvalidRunStatusError as e:
-            error_message = f"Cannot visualize non-completed run. {str(e)}"
-            logger.error(error_message)
+
+        # if there exists flex flow run, use trace UI to visualize
+        # maybe we can fully switch to trace UI for DAG flow run in the future
+        has_flex_flow_run = any([run._flow_type == FlowType.FLEX_FLOW for run in validated_runs])
```

Comment:
will prompty run go this branch?

Reply:
currently no, but i think prompty should also go into this - let me update this

Reply:
updated, may take a look